### DISCUSSION TITLE
Nurbs support in Pixi

### DISCUSSION THREAD

**Discussion by Cuberice** [pixijs/pixijs#9265] [Q&A]
We are using Pixijs for rendering in our application. We have a request to implement a large amount of nodes and edges using NURBS. 

I see that webGL has some support for nurbs, and Three.js as well, but I am not really able to find anything for PixiJs. Is there support build into Pixi for calculating and drawing edges with Nurbs?

I have also found this library but it does not seem like it is being maintained: [verb-nurbs-web](https://www.npmjs.com/package/verb-nurbs-web)

**Comment by SuperSodaSea** [comment]
WebGL don't have direct support for NURBS, it only supports basic primitives like points, lines and triangles. Three.js just converts NURBS to lines and triangles and then render it, see [webgl_geometry_nurbs.html](https://github.com/mrdoob/three.js/blob/47b28bc564b438bf2b80d6e5baf90235292fcbd7/examples/webgl_geometry_nurbs.html), [NURBSCurve.js](https://github.com/mrdoob/three.js/blob/47b28bc564b438bf2b80d6e5baf90235292fcbd7/examples/jsm/curves/NURBSCurve.js), [NURBSSurface.js](https://github.com/mrdoob/three.js/blob/47b28bc564b438bf2b80d6e5baf90235292fcbd7/examples/jsm/curves/NURBSSurface.js) and [NURBSUtils.js](https://github.com/mrdoob/three.js/blob/47b28bc564b438bf2b80d6e5baf90235292fcbd7/examples/jsm/curves/NURBSUtils.js) in Three.js example. You can try the code in NURBSUtils.js above to calculate the coordinates of the points and draw the lines in PixiJS to form NURBS.

**Comment by Cuberice** [comment]
We used a package called b-spline which works quite well
